src/httpServer/server.ts

```diff
@@ -61,18 +62,48 @@ export class DBOSHttpServer {
    * Register HTTP endpoints and attach to the app. Then start the server at the given port.
    * @param port
    */
-  listen(port: number) {
-    // Start the HTTP server.
+  async listen(port: number) {
+    try {
+      await this.checkPortAvailability(port, "127.0.0.1");
+      await this.checkPortAvailability(port, "::1");
+    } catch (error) {
+      this.logger.warn(`Port ${port} is already used. Please use the -p option to choose another port.`);
+      process.exit(1);
+    }
+
     const appServer = this.app.listen(port, () => {
       this.logger.info(`DBOS Server is running at http://localhost:${port}`);
     });
+
     const adminPort = port + 1
     const adminServer = this.adminApp.listen(adminPort, () => {
       this.logger.info(`DBOS Admin Server is running at http://localhost:${adminPort}`);
     });
     return {appServer: appServer, adminServer: adminServer}
   }
 
+async checkPortAvailability(port: number, host: string): Promise<void> {
```

Comment:
Do we have to create a new temporary server?  Can we not get this information out of `appServer`?

Reply:
alternative solution: https://github.com/facebook/docusaurus/blob/0589b1475d56b0b541348aa56f201ec7c56c56d5/packages/docusaurus/src/server/getHostPort.ts#L25